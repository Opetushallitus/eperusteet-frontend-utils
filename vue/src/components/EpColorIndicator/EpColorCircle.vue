<template>
<span ref="circle"
      :style="circleStyle"
      :title="$t(help)"
      :class="circleClass">
  <b-popover v-if="help"
             :target="() => $refs['circle']"
             :placement="'top'"
             triggers="hover"
             variant="primary">
    <span>{{$t(help)}}</span>
  </b-popover>
</span>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'vue-property-decorator';

@Component
export default class EpColorCircle extends Vue {
  @Prop({ default: '#000000' })
  color!: string;

  @Prop({ required: false })
  help!: string | undefined;

  @Prop({ default: 10 })
  size!: number;

  get circleStyle() {
    return {
      'min-height': this.size + 'px',
      'min-width': this.size + 'px',
      'background': this.color,
    };
  }

  get circleClass() {
    return this.help ? 'circle circle-tooltip' : 'circle';
  }
}
</script>

<style lang="scss" scoped>
@import '../../styles/_variables.scss';

.circle {
  background: black;
  border-radius: 100%;
  display: inline-block;
}

.circle-tooltip {
  cursor: help;
}

</style>
