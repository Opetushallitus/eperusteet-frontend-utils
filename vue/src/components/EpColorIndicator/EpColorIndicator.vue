<template>
  <span ref="ball"
        class="material-icons"
        :style="dynstyle"
        :title="$t(kind)"
        :class="spanClass"
        aria-hidden="true">
        circle
    <b-popover v-if="tooltip"
               :target="() => $refs['ball']"
               :placement="'top'"
               triggers="hover"
               variant="primary">
      <span>{{$t(kind)}}</span>
    </b-popover>
  </span>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'vue-property-decorator';
import { themeColors, themes, rgb2string } from '../../utils/perusteet';

const moduuliColors = {
  'normaali': [0, 0, 0],
  'pakollinen': [189, 219, 138],
  'valinnainen': [241, 102, 192],
  'paikallinen': [255, 165, 0],
  'valittu': [128, 255, 192],
};

export type IndicatorKind = 'normaali'
  | 'pakollinen'
  | 'valinnainen'
  | 'laadinta'
  | 'julkaistu'
  | 'valmis'
  | 'poistettu'
  | 'paikallinen'
  | 'offline'
  | 'online'
  | 'ammatillinen'
  | 'esiopetus'
  | 'lukiokoulutus'
  | 'perusopetus'
  | 'varhaiskasvatus'
  | 'taiteenperusopetus'
  | 'vapaasivistystyo'
  | 'tutkintoonvalmentava'
  | 'kotoutumiskoulutus';

@Component
export default class EpColorIndicator extends Vue {
  @Prop({ default: 'normaali' })
  kind!: IndicatorKind;

  @Prop({ default: true })
  tooltip!: boolean;

  @Prop({ default: 10 })
  size!: number;

  @Prop({ required: false })
  backgroundColor!: string;

  get spanClass() {
    return `ball ball-${this.kind}${this.tooltip ? ' ball-tooltip' : ''}`;
  }

  get background() {
    return themeColors[themes[this.kind]] || themeColors[this.kind] || moduuliColors[this.kind] || [0, 0, 0];
  }

  get dynstyle() {
    const result = {
      'color': this.backgroundColor ? this.backgroundColor : rgb2string(this.background),
    };

    return result;
  }
}
</script>

<style lang="scss" scoped>
@import '../../styles/_variables.scss';

.ball {
  font-size: 0.7rem;
  color: black;
}

.ball-normaali {
  color: #000000;
}

.ball-pakollinen {
  color: #bddb8a;
}

.ball-valinnainen {
  color: #f166c0;
}

.ball-paikallinen {
  color: #ffa500;
}

.ball-offline {
  color: #DADADA;
}

.ball-online {
  color: #7CD443;
}

.ball-tooltip {
  cursor: help;
}

</style>
