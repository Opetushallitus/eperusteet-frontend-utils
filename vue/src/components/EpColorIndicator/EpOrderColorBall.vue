<template>
  <span :style="'color:' + kohdealueVarit[index]">
    <EpMaterialIcon size="14px">circle</EpMaterialIcon>
  </span>
</template>

<script lang="ts">

import { Component, Vue, Prop } from 'vue-property-decorator';
import EpMaterialIcon from '@shared/components/EpMaterialIcon/EpMaterialIcon.vue';

@Component({
  components: {
    EpMaterialIcon,
  },
})
export default class EpOrderColorBall extends Vue {
  @Prop({ required: true })
  private index!: number;

  private kohdealueVarit = ['#99B3F1', '#9BDCFF', '#B2B2B2', '#002D99', '#FFD900', '#E60895', '#C126B8', '#575757',
    '#2e3192', '#8283be', '#663300', '#a38566', '#666600', '#a3a366'];
}
</script>

<style lang="scss" scoped>
</style>
