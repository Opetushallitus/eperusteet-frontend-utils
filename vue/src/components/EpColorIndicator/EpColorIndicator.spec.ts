import { mount, createLocalVue, shallowMount } from '@vue/test-utils';
import EpColorIndicator from './EpColorIndicator.vue';
import Vue from 'vue';
import BootstrapVue from 'bootstrap-vue';

Vue.use(BootstrapVue);

describe('EpToggle component', () => {
  const localVue = createLocalVue();

  test('Renders toggle and change changes value', async () => {
    const wrapper = mount(EpColorIndicator, {
      localVue,
      mocks: {
        $t: x => x,
      },
      propsData: {},
    });

    expect(wrapper.html()).toContain('normaali');

    wrapper.setProps({
      kind: 'julkaistu',
      tooltip: 'tooltipviesti',
      size: 20,
    });

    expect(wrapper.html()).toContain('julkaistu');
    expect(wrapper.html()).toContain('circle');

    expect(wrapper.element.style.color).toMatchSnapshot();

    wrapper.setProps({
      kind: 'esiopetus',
      tooltip: 'tooltipviesti',
      size: 20,
    });

    expect(wrapper.element.style.color).toMatchSnapshot();
  });
});
