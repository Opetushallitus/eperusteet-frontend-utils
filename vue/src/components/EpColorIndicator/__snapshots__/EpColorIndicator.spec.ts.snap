// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpToggle component Renders toggle and change changes value 1`] = `"rgb(0, 0, 0)"`;

exports[`EpToggle component Renders toggle and change changes value 2`] = `"rgb(132, 210, 255)"`;
